supervision/detection/line_counter.py

```diff
@@ -197,6 +201,69 @@ def __init__(
         self.text_padding: int = text_padding
         self.custom_in_text: str = custom_in_text
         self.custom_out_text: str = custom_out_text
+        self.display_in_count: bool = display_in_count
+        self.display_out_count: bool = display_out_count
+
+    def _annotate_count(
+        self,
+        frame: np.ndarray,
+        line_counter: LineZone,
+        text: str,
+        in_count: bool,
+    ) -> None:
+        """This method is drawing the text on the frame.
+
+        Args:
+            frame (np.ndarray): The image on which the text will be drawn.
+            line_counter (LineCounter): The line counter
+                that will be used to draw the line.
+            text (str): The text that will be drawn.
+            in_count (bool): Whether to display the in count or not.
+        """
+        (text_width, text_height), _ = cv2.getTextSize(
+            text, cv2.FONT_HERSHEY_SIMPLEX, self.text_scale, self.text_thickness
+        )
+
+        text_x = (
+            line_counter.vector.end.x + line_counter.vector.start.x - text_width
+        ) / 2
+
+        text_y = (
+            line_counter.vector.end.y + line_counter.vector.start.y + text_height
+        ) / 2
+
+        if in_count:
+            # if the text is an in count, we want to display it above the line
+            text_y -= self.text_offset * text_height
+        else:
+            # if the text is an out count, we want to display it below the line
+            text_y += self.text_offset * text_height
+
+        text_background_rect = Rect(
+            x=text_x,
+            y=text_y - text_height,
+            width=text_width,
+            height=text_height,
+        ).pad(padding=self.text_padding)
+
+        cv2.rectangle(
```

Comment:
We could use `draw_text` here and remove the logic for drawing text and background. We already use it to draw the `PolygonZone` label.

Reply:
Thanks for your advice. I updated my PR.  I needed to make some minor updates to the draw_text function. I'm open to different suggestions